sys/linux/socket_inet6.txt

```diff
@@ -189,3 +189,44 @@ in6_ifreq {
 	ifr6_prefixlen	int32[0:128]
 	ifr6_ifindex	ifindex
 }
+
+# sendmsg for ip_cmsg_send, ip6_datagram_send_ctl
+
+msghdr_inet6 {
+	msg_name	ptr[in, sockaddr_in6, opt]
+	msg_namelen	len[msg_name, int32]
+	msg_iov		ptr[in, array[iovec_in]]
+	msg_iovlen	len[msg_iov, intptr]
+	msg_control	ptr[in, array[cmsghdr_inet6], opt]
+	msg_controllen	bytesize[msg_control, intptr]
+	msg_flags	const[0, int32]
+}
+
+mmsghdr_inet6 {
+	msg_hdr	msghdr_inet6
+	msg_len	const[0, int32]
+}
+
+ipv6_opt_hdr {
+	nexthdr	int8
+	hdelen	int8
+}
```

Comment:
This struct is missing option data, which is different for each particular option. Either add proper descriptions for all options types or a `data array[int8]` filler and a TODO that this is not implemented.

Reply:
I found there is ipv6_dstopts_ext_header/ipv6_hopopts_ext_header which can be used for  hopopts/dstopts. ipv6_opt_hdr seems is not need.